packages/openapi-tanstack-query/test/fixtures/mock-server.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  http,
+  HttpResponse,
+  type JsonBodyType,
+  type StrictRequest,
+  type DefaultBodyType,
+  type HttpResponseResolver,
+  type PathParams,
+  type AsyncResponseResolverReturnType,
+} from "msw";
+import { setupServer } from "msw/node";
+
+/**
+ * Mock server instance
+ */
+export const server = setupServer();
+
+/**
+ * Default baseUrl for tests
+ */
+export const baseUrl = "https://api.example.com" as const;
+
+/**
+ * Test path helper, returns a an absolute URL based on
+ * the given path and base
+ */
+export function toAbsoluteURL(path: string, base: string = baseUrl) {
+  // If we have absolute path
+  // if (URL.canParse(path)) {
+  //   return new URL(path).toString();
+  // }
+
+  // Otherwise we want to support relative paths
+  // where base may also contain some part of the path
+  // e.g.
+  // base = https://api.foo.bar/v1/
+  // path = /self
+  // should result in https://api.foo.bar/v1/self
+
+  // Construct base URL
+  const baseUrlInstance = new URL(base);
+
+  // prepend base url url pathname to path and ensure only one slash between the URL parts
+  const newPath = `${baseUrlInstance.pathname}/${path}`.replace(/\/+/g, "/");
+
+  return new URL(newPath, baseUrlInstance).toString();
+}
+
+export type MswHttpMethod = keyof typeof http;
+
+export interface MockRequestHandlerOptions<
+  // Recreate the generic signature of the HTTP resolver
+  // so the arguments passed to http handlers propagate here.
+  Params extends PathParams<keyof Params> = PathParams,
+  RequestBodyType extends DefaultBodyType = DefaultBodyType,
+  ResponseBodyType extends DefaultBodyType = undefined,
+> {
+  baseUrl?: string;
+  method: MswHttpMethod;
+  /**
+   * Relative or absolute path to match.
+   * When relative, baseUrl will be used as base.
+   */
+  path: string;
+  body?: JsonBodyType;
+  headers?: Record<string, string>;
+  status?: number;
+
+  /**
+   * Optional handler which will be called instead of using the body, headers and status
+   */
+  handler?: HttpResponseResolver<Params, RequestBodyType, ResponseBodyType>;
+}
+
+/**
+ *  Configures a msw request handler using the provided options.
+ */
+export function useMockRequestHandler<
```

Comment:
Nice! I like this “hook-like” setup for the React tests. Makes sense.

Reply:
Literally copy pasted from here https://github.com/openapi-ts/openapi-typescript/blob/7c7d7153007b48c4c69eeeadebf8a4817f473a4a/packages/openapi-fetch/test/fixtures/mock-server.ts#L78

I try my best to adapt my way of writing based on the original repository so we don't have codes that look completely different between each library